File: german-telework/service/tasks/TransportCrypt.java
import java.util.*;


public class TransportCrypt {

    long initialSeed = 0xEA0A8EEA0A8EEA0AL;
    long mask = 0xffffffffffffffffL;

    long seed;
    byte count;

    public TransportCrypt(){
        this.seed = initialSeed;
        this.count = 0;
    }

    private long transportCryptKeystream(long seed){
        long result = seed;
        result ^= result << 13 & mask;
        result ^= result >>> 7 & mask;
        result ^= result << 17 & mask;
        return result & mask;
    }

    private byte getKeyByte(){
        if (this.count == 16){
            this.count = 0;
            this.seed = this.transportCryptKeystream(this.seed);
        }
        byte result = (byte) ((this.seed >> this.count) & 0xff);
        this.count += 1;
        return result;
    }

    private byte[] transportCryptState(byte[] buffer){
        List<Byte> tmp = new ArrayList<Byte>();
        for (byte b : buffer) {
            byte keyByte = this.getKeyByte();
            tmp.add((byte) (b ^ keyByte));
        }
        byte[] res = new byte[tmp.size()];
        for (int i = 0; i<tmp.size(); i++)
            res[i] = tmp.get(i);
        return res;
    }

    public static byte[] cryptoBuffer(byte[] buffer) {
        TransportCrypt tc = new TransportCrypt();
        return tc.transportCryptState(buffer);
    }
}

File: german-telework/service/holiday/Holiday/TransportCrypt.cs
public class TransportCrypt {

    ulong InitialSeed = 0xEA0A8EEA0A8EEA0A;
    ulong Mask = 0xffffffffffffffff;

    ulong Seed;
    byte Count;

    public TransportCrypt(){
        this.Seed = InitialSeed;
        this.Count = 0;
    }

    private ulong TransportCryptKeystream(ulong seed){
        ulong result = seed;
        result ^= result << 13 & Mask;
        result ^= result >> 7 & Mask;
        result ^= result << 17 & Mask;
        return result & Mask;
    }

    private byte GetKeyByte(){
        if (this.Count == 16){
            this.Count = 0;
            this.Seed = this.TransportCryptKeystream(this.Seed);
        }
        byte result = (byte) ((this.Seed >> this.Count) & 0xff);
        this.Count += 1;
        return result;
    }

    private byte[] TransportCryptState(byte[] buffer){
        List<Byte> tmp = new List<Byte>();
        foreach (byte b in buffer) {
            byte keyByte = this.GetKeyByte();
            tmp.Add((byte) (b ^ keyByte));
        }
        byte[] res = tmp.ToArray();
        return res;
    }

    public static byte[] CryptoBuffer(byte[] buffer) {
        TransportCrypt tc = new TransportCrypt();
        return tc.TransportCryptState(buffer);
    }
}
